src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -199,14 +203,28 @@ impl TargetInfo {
                 process.arg("--crate-type").arg(crate_type.as_str());
             }
 
-            // An extra `rustc` call to determine `-Csplit-debuginfo=packed` support.
-            let supports_split_debuginfo = rustc
+            // 3 extra `rustc` calls to determine `-Csplit-debuginfo={off,packed,unpacked}` support.
```

Comment:
Your patch indeed works. However, while I won't say they are excessive, two more `rustc` calls might make the [already bad startup time](https://github.com/rust-lang/cargo/issues/10986) of Cargo worse. I wonder if we could have an alternative way to solve it.

Off the top of my head, here are two options:

* Add a new `rustc --print` value such as `rustc --print=split-debuginfo`, so that we only need to call it once per target platform. The stabilized status is [queried here in rustc](https://github.com/rust-lang/rust/blob/09508489efc223287731fe8abbd2a81bbf7adf8e/compiler/rustc_session/src/session.rs#L1545-L1549).
* The other way around is providing a hardcoded value for already stabilized options. That means we need to do something like `triple.ends_with("-windows-gnu")` to differentiate each platform ([like this one](https://github.com/rust-lang/cargo/blob/df56877805e613969da837e1ede28fda3a17481d/src/cargo/core/compiler/build_context/target_info.rs#L368)).

I am not strongly against or prefer to any of them, though.

Reply:
I agree that limiting startup times would be beneficial, especially in the context of CI which has brought this issue to my attention.

I've created a PR implementing the suggested `--print` approach in the main Rust repository: https://github.com/rust-lang/rust/pull/104104, I believe it will require less maintenance that checking the target via `triple.ends_with("-windows-gnu")` etc.

Reply:
I'll wait with updating this PR until I get some feedback on my Rust PR, in case it's rejected or different UX is expected.

Reply:
As my rustc PR got approved, I've rewritten this patch to reflect that.